core/state/statedb.go

```diff
@@ -1484,3 +1486,13 @@ func (s *StateDB) PointCache() *utils.PointCache {
 func (s *StateDB) Witness() *stateless.Witness {
 	return s.witness
 }
+
+func (s *StateDB) IsSlotFilled(addr common.Address, slot common.Hash) bool {
+	// The snapshot can not be used, because it uses the old encoding where
+	// no difference is made between 0 and no data.
+	_, err := s.db.DiskDB().Get(utils.StorageSlotKeyWithEvaluatedAddress(s.accessList.pointCache.GetTreeKeyHeader(addr[:]), slot[:]))
```

Comment:
should go through the reader abstaction, because it might have been written in a previous block but not in the db / also we want to use the layers to build the witness if we can.

Reply:
@rjl493456442  `s.reader.StorageExist(addr, slot)`